CalendarApp/GoogleCalendarService/Manager/EventPeriodicityHolder.cs

```diff
@@ -0,0 +1,143 @@
+                Regex regex = new Regex(pattern);
+                Match match = regex.Match(eventForAdd.Recurrence[0]);
+
+                Frequency = match.Groups["FREQ"].Value;
+                if(match.Groups["BYDAY"].Value != "")
+                    SetDayOfWeeksList(match.Groups["BYDAY"].Value);
+                if (match.Groups["COUNT"].Value != "")
+                {
+                    FreqType = FrequencyType.COUNT;
+                    Count = Convert.ToInt32(match.Groups["COUNT"].Value);
+                }
+                if (match.Groups["UNTIL"].Value != "")
+                {
+                    FreqType = FrequencyType.UNTIL;
+                    UntilDateTime = GetUntilDateTime(match.Groups["UNTIL"].Value);
+                }
+            }
+
+            if(Frequency == "DAILY")
+                EventList = EventsListBuilder.StartConstruction().WithDailyEvents(this).Build();
+            else if(Frequency == "WEEKLY")
+                EventList = EventsListBuilder.StartConstruction().WithWeeklyEvents(this).Build();
+            else if (Frequency == "MONTHLY")
+                EventList = EventsListBuilder.StartConstruction().WithMonthlyEvents(this).Build();
+            else if (Frequency == "YEARLY")
+                EventList = EventsListBuilder.StartConstruction().WithYearlyEvents(this).Build();
+        }
+
+        private DateTime GetUntilDateTime(string untilDate)
+        {
+            string pattern = @"(?<Year>\d{4})(?<Month>\d{2})(?<Day>\d{2})T(?<Hour>\d{2})(?<Minute>\d{2})(?<Second>\d{2})Z";
+            Regex regex = new Regex(pattern);
+            Match match = regex.Match(untilDate);
+
+            var hour = match.Groups["Hour"].Value == "00" ? 0 : Convert.ToInt32(match.Groups["Hour"].Value);
+            var minute = match.Groups["Minute"].Value == "00" ? 0 : Convert.ToInt32(match.Groups["Minute"].Value);
+            var second = match.Groups["Second"].Value == "00" ? 0 : Convert.ToInt32(match.Groups["Second"].Value);
+
+            return new DateTime(
+                Convert.ToInt32(match.Groups["Year"].Value),
+                Convert.ToInt32(match.Groups["Month"].Value),
+                Convert.ToInt32(match.Groups["Day"].Value),
+                hour,
+                minute,
+                second
+                );
+        }
+
+        private void SetDayOfWeeksList(string byDay)
+        {
+            DayOfWeeksList = byDay.Split(',').Select(ExtractDay).ToList();
+            WeekEventList = DayOfWeeksList.Select(ExtractEventTime).ToList();
+        }
+
+        private Event ExtractEventTime(DayOfWeek dayOfWeek)
+        {
+            DayOfWeek realEventDay = RealEvent.Start.DateTime.Value.DayOfWeek;
+            Event newEvent = null;
+            int day;
+            if (realEventDay == DayOfWeek.Sunday)
+                day = 7;
+            else
+                day = (int) realEventDay;
+
+            DateTime weekDateTime = RealEvent.Start.DateTime.Value.AddDays(-day+1);
+
```

Comment:
In case of Sunday this would be -7 +1 = -6. And all events would be added to the list starting from previous week? Am I missing something?